okhttp/src/main/java/okhttp3/RealCall.java

```diff
@@ -97,7 +97,11 @@ private void captureCallStackTrace() {
     }
     captureCallStackTrace();
     eventListener.callStart(this);
-    client.dispatcher().enqueue(new AsyncCall(responseCallback));
+    try {
+      client.dispatcher().enqueue(new AsyncCall(responseCallback));
+    } catch (Exception e) {
+      responseCallback.onFailure(RealCall.this, new IOException(e.getMessage()));
```

Comment:
Can you put a message, and chain the original exception e.g.

`new IOException("unable to enqueue to dispatcher", e);`